### ISSUE TITLE
Parsing error when trying to assign list variable

### ISSUE THREAD

**Issue by tappleby** [osohq/oso#389]
When trying to assign a variable in a rule where one list item references an object property, I am running into a syntax parsing error "did not expect to find the token ']' at line X"

```polar
resource_scope(actor: Person, "read", "Person", filters) if
    filters = ["id", "=", actor.id];
```

The same rule works fine if assigning a dictionary instead of list:

```polar
resource_scope(actor: Person, "read", "Person", filters) if
    filters = { id: actor.id };
```

As a workaround, the following will parse correctly:

```polar
resource_scope(actor: Person, "read", "Person", filters) if
    field = actor.id and
    filters = ["id", "=", field];
```

**Comment by samscott89** [comment]
Thank you for finding and opening an issue!

The issue looks like a parsing error: currently expressions aren't valid inside lists. So similarly `x = [1+1]` also fails. We will have a fix up soon!

**Comment by samscott89** [comment]
Thanks again for opening this issue. The fix was including in v0.5.1
